### PR TITLE
[WIP] add scalajs-react

### PR DESCRIPTION
**Pull request by SethTisue**
references #798.  /cc @sjrd @japgolly

### CODE REVIEW

**Comment by SethTisue** [comment]
queued: 
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3626/consoleFull

**Comment by SethTisue** [comment]
in local test, the scalajs-react "core" project builds, but then when it tries to build "extra",

```
[scalajs-react] [warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[scalajs-react] [warn] 	::          UNRESOLVED DEPENDENCIES         ::
[scalajs-react] [warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[scalajs-react] [warn] 	:: com.github.japgolly.scalajs-react#core;1.3.2-dbuildx0ea4fff6fa5190a90a767f4c9af744e43fc61d0e: not found
[scalajs-react] [warn] 	::::::::::::::::::::::::::::::::::::::::::::::
```

I saw the same pattern in scala-js-dom, where building "root" works, but trying to build "extra" fails to find the dependency on the thing "root" published

@cunei it's tantalizing, it seems like this might be thiiiiiiiis close to working. any clue what's going on here?

**Comment by cunei** [comment]
Hi @SethTisue, I investigated the matter a while back. If I remember correctly, the issue is in the generation of version numbers by Scala.js, which appends to the version some additional string in order to distinguish the JavaScript artifacts from the Java ones; I seem to remember that classifiers are involved, but I might be mistaken. While changing version numbers, dbuild does not know that the suffix is supposed to be part of the artifact name, and builds the artifact id in a manner that is not the same that Scala.js expects. Adding the needed support for Scala.js in dbuild is feasible, but requires handling Scala.js artifact names in an ad-hoc manner, meaning that some more code is needed.
Adding Scala.js support has not been a priority for the past few years; if you think that adding support would be important, you should ask for some development time to be devoted to that. A fair estimate would one to two weeks.

**Comment by japgolly** [comment]
Hey @SethTisue, thanks for setting this up. LGTM

File: configs/community.dbuild [hunk 1]
```diff
@@ -411,1940 +222,28 @@ build += {
...
-    name: "circe"
-    uri:  ${vars.uris.circe-uri}
-    extra.sbt-version: ${vars.sbt-1-version}
-    extra.projects: [
-      // easy
-      "coreJVM", "numbersJVM"
-      // harder
-      "jawn"
-      // bunch more stuff that all depends on jawn
-      "parserJ
```
Comment (japgolly) [inline configs/community.dbuild:246]: Can I ask, what's the reason for excluding these? Is it to save build time?

Comment (SethTisue) [reply configs/community.dbuild:246]: this is an include list, not an exclude list. `extra.projects` is include, `extra.exclude` is exclude.

typically we include what downstream projects actually depend on, and then the rest is gravy (we include or not based on some estimate of cost/benefit. most often the cost has to do with maintenance effort, not build time)

Comment (SethTisue) [reply configs/community.dbuild:246]: (btw I'm always happy, if someone asks, to add a comment to the config file clarifying a decision of this sort. you'll see there are already many such comments, but also many places where they are missing.)

Comment (japgolly) [reply configs/community.dbuild:246]: Ok cool thanks for clarifying. Can I confirm also that `test:compile` is run on the included projects? If it is then exclusion of the `gh-pages` module here is fine but if not; well preferably we'd switch it on for scalajs-react but if that's unfeasible for whatever reason then enabling `gh-pages` module would be the second-most valuable thing.

Comment (SethTisue) [reply configs/community.dbuild:246]: > Can I confirm also that `test:compile` is run on the included projects?

not only that, but `test:test`, too

Comment (japgolly) [reply configs/community.dbuild:246]: Awesome! Cheers

**Comment by smarter** [comment]
> if you think that adding support would be important, you should ask for some development time to be devoted to that. A fair estimate would one to two weeks.

Is there anything that non-Lightbend people can do to encourage/help prioritize that ? Do +1 on https://github.com/scala/community-builds/issues/798 count ? I believe that this would be well worth the effort.

**Comment by SethTisue** [comment]
> Is there anything that non-Lightbend people can do to encourage/help prioritize that ? Do +1 on #798 count ? I believe that this would be well worth the effort.

the one thing I can definitely say, that I've said in public a couple times before, is that any current or potential Lightbend customers with a commercial interest in Scala.js should definitely communicate that interest to the business and/or sales sides of Lightbend.

whether there are _other_ avenues by which this could be encouraged and helped, I really just don't know, it's above my pay grade, so to speak. regardless, I'd personally love to see this happen and I remain optimistic about it (as, admittedly, I have been for a long time now).

closing for now, just because I don't like to leave pull requests open indefinitely.

@cunei would it be worth asking @sjrd if he sees some way to work around this? I'm not clear enough on the artifact naming issue to know whether the suggestion even makes sense.

it would be valuable even if the workaround only worked for projects immediately downstream from Scala.js, we could worry later about whether it supported, for example, including projects which (for example) depended on scalajs-react

**Pull request closed without merging**